dandi/files/bids.py

```diff
@@ -61,51 +65,78 @@ def bids_root(self) -> Path:
         """
         return self.filepath.parent
 
-    def _validate(self) -> None:
+    @property
+    def bids_version(self) -> str | None:
+        """
+        The version of BIDS used for in validation
+
+        Note
+        ----
+            This value is not necessarily the same as the value of the `"BIDSVersion"`
+            field in the represented `dataset_description.json` file.
+        """
+        self._validate()
+        return self._bids_version
+
+    def _get_metadata(self) -> None:
+        """
+        Get metadata for all assets in the dataset
+
+        This populates `self._asset_metadata`
+        """
         with self._lock:
-            if self._dataset_errors is None:
+            if self._asset_metadata is None:
                 # Import here to avoid circular import
                 from dandi.validate import validate_bids
```

Comment:
Oh my, OK so to my understanding, `_get_metadata` still relies on the old `dandi.validate.validate_bids`?

Used by `get_asset_metadata` at https://github.com/dandi/dandi-cli/pull/1599/files#diff-70a4bb759311189af6f4468d78254d863319fc4660ebe437060b6634c5a7b724L106

And now new `_validate` method (https://github.com/dandi/dandi-cli/pull/1599/files#diff-70a4bb759311189af6f4468d78254d863319fc4660ebe437060b6634c5a7b724R109) called by previous (and still) `get_validation_errors` at line https://github.com/dandi/dandi-cli/pull/1599/files#diff-70a4bb759311189af6f4468d78254d863319fc4660ebe437060b6634c5a7b724R151 is the 'replacement' meant here

Reply:
Is that an apt summary of the 'integration' of the new feature?

Reply:
> Oh my, OK so to my understanding, `_get_metadata` still relies on the old `dandi.validate.validate_bids`?

Yes. There is no equivalent metadata provided by the deno-compiled validator, so we are keeping the `bidsschematools` to provide the same kind of metadata.

> Is that an apt summary of the 'integration' of the new feature?

Yes, that's the gist of it. I found the current design of validation (metadata gathering as well) in DANDI complex. This is a method of integration that I found that adhere to the current design. 

@yarikoptic may have some plan for the validation process in dandi-cli in the future. What I see is that any design change in the validation process would entail significant changes in other parts of the program.